### PR TITLE
5106 Implements validation in device form

### LINKED ISSUE #5106: Add form validation for support admin device forms
Currently, anyone with access to the support admin dashboard can add and edit devices that contain empty values for required fields.

**Steps to repro:**
On the add devices or edit devices as a support admin, you can edit or add a device. If you type and then delete a required field like "Device name" then the "Save changes" button is enabled. You can then submit a device with no fields filled out to be added.

**Action requested**
Prevent the user from submitting empty values for required fields.

### LINKED ISSUE #5184: Add "note" role to Reminder on add/edit device page
Came out of the discussion [here](https://github.com/CDCgov/prime-simplereport/pull/5177#discussion_r1092361932) in a recent PR - would be helpful to add the "note" role to the Reminder section.

### Acceptance criteria
The reminder message when adding/editing devices should have a note role

### Notes
**Open question**: do we want a specific alert component? or just to add alert role to this component?
**Answer**: we will just add "note" role to this element

### PR DESCRIPTION
**Pull request by johanna-skylight**
# FRONTEND PULL REQUEST

## Related Issue

- Resolves #5106
- Resolves #5184 

## Changes Proposed
- Adds react-hook-forms library to the SimpleReport project. This library will take care of performing basic validations, keep track of the form data state, trigger errors accordingly and perform the focus on error on elements that are properly integrated
- Modifies form common components  `Dropdown, Select, TextInput, MultiSelect` so they can be properly setup with the library.
- Updates the deviceForm component to use `react-hook-form` to track data state, errors and validation. It also removes the prop initialValues as it was not really used.
- Moves the logic of the `disease information` section into its own component.
- Updates necessary unit testing.
- Updates the container components `ManageDeviceTypeFormContainer` and `DeviceTypeFormContainer`
- Adds the role attribute to the component `DeviceTypeReminderMessage`
- Updates the dom references of the multiplex e2e spec 09.

## Additional Information

I ended up modifiying the `MultiSelect` component and the underlying components because the workaround had small glitches.

## Testing
_Tested in Dev1_

- Verify that all fields marked as required must be filled out in order to submit the form.
- Verify that if the field is empty the form should display an error triggered but the click of the `save changes` button.
- Verify that `save changes` button enables when the form is "dirty".
- Verify that `test length` field only allows for numbers and also displays an error if a number outside the 0-999 range is entered.
- Verify that the http request to update the data only triggers if the form has been filled out with valid values.
- Edit mode should load values as expected upon selecting a device. 

## Screenshots / Demos
Screenshot 2023-03-07 at 4 11 41 PM

### CODE REVIEW

**Comment by emyl3** [comment]
One weird bug I'm noticing is I type one number in "Test length (minutes)" field and it automatically enters to the next field... Can you repro this?

For some reason I can't attach the screen recording so I'll put it in slack -_-"

File: frontend/src/app/supportAdmin/DeviceType/DiseaseInformation.tsx [hunk 1]
```diff
@@ -0,0 +1,168 @@
+import React from "react";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import {
+  Control,
+  FieldErrors,
+  useFieldArray,
```
Comment (emyl3) [inline frontend/src/app/supportAdmin/DeviceType/DiseaseInformation.tsx:6]: cool! https://react-hook-form.com/api/usefieldarray/ 👀

**Review by emyl3 — COMMENTED** [review]
Thank you for introducing the new library and offering a more robust solution.

I only noticed one thing where on the `<MultiSelect>` component doesn't receive the focus the first time when trying to submit/update the device form.

Once again, can't add to github so adding to slack thread ^__^"

**Review by BobanL — APPROVED** [review]
Looks great! Thank you for adding validation here!

**Comment by johanna-skylight** [comment]
@zdeveloper I remember you telling you were the one that created the pill component. I have aded some small logic to integrate it with react-hook-forms if you spot any issues with its behavior please let me know!. I tested the component against the current version in main and it seems that is working as expected... but an extra set of eyes is always good.

File: frontend/src/app/commonComponents/Select.tsx [hunk 2]
```diff
@@ -23,6 +24,7 @@ interface Props<T> {
   selectClassName?: string;
   disabled?: boolean;
   className?: string;
+  registrationProps?: UseFormRegisterReturn<any>;
```
Comment (fzhao99) [inline frontend/src/app/commonComponents/Select.tsx:27]: for my understanding: what would go into typing the generic passed into `UseFormRegisterReturn` more explicitly (ie not using `any` here)? Vaguely understand that the set of input rules for the relevant components would be different and therefore difficult to type more explicitly, but trying to grok how the type system is interacting with the props here.

**Review by BobanL — APPROVED** [review]

**Review by emyl3 — APPROVED** [review]
LGTM! Thank you for all your work on this!! 😹

**Pull request merged**